doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,483 @@
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_attr_t;
+
+/**
+ * @brief Local SID Entry
+ */
+typedef struct _sai_local_sid_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief Length of the Locator part of the SID
+     */
+    sai_uint8_t locator_len;
+
+    /**
+     * @brief Length of the Function part of the SID
+     */
+    sai_uint8_t function_len;
+
+    /**
+     * @brief Length of the Args part of the SID
+     */
+    sai_uint8_t args_len;
+
+    /**
+     * @brief Local Segment
+     */
+    sai_ip6_t sid;
+
+} sai_local_sid_entry_t;
+
+```
+
+5. Added apis to create/remove and set/get attributes for local_sid_entry
+
+    - add_local_sid_entry
+    - remove_local_sid_entry
+    - set_local_sid_entry_attribute
+    - get_local_sid_entry_attribute
+    - create_local_sid_entries
+    - remove_local_sid_entries
+    - set_local_sid_entries_attribute
+    - get_local_sid_entries_attribute
+
+
+## Examples ##
+- SR Headend
+    Example configuration for H.Encaps.Red behavior
+       
+        1. Create a SID list object with 3 segments
+
+            sidlist_entry_attrs[0].id = SAI_SEGMENTROUTE_SIDLIST_ATTR_TYPE
+            sidlist_entry_attrs01].value.s32 = SAI_SEGMENTROUTE_SIDLIST_TYPE_ENCAPS_RED
+            sidlist_entry_attrs[1].id = SAI_SEGMENTROUTE_SIDLIST_ATTR_SEGMENT_LIST
+            sidlist_entry_attrs[1].value.objlist.count = 3;
+            CONVERT_STR_TO_IPV6(sidlist_entry_attrs[1].value.objlist.list[0], "2001:db8:85a3::8a2e:370:7334");
+            CONVERT_STR_TO_IPV6(sidlist_entry_attrs[1].value.objlist.list[1], "2001:db8:85a3::8a2e:370:2345");
+            CONVERT_STR_TO_IPV6(sidlist_entry_attrs[1].value.objlist.list[2], "2001:db8:85a3::8a2e:370:3456");
+            saistatus = sai_v6sr_api->create_segmentroute_sidlist(&sidlist_id, switch_id, 2, sidlist_entry_attrs);    
```

Comment:
do we have/need query to find the capability of how many v6 sid can be encapsulated ?

Reply:
In SRv6 workgroup meeting, we agreed to implement additional attributes for length of SID list in a subsequent PR. Marking this as resolved for this PR.